lib/javascript/JavascriptParser.js

```diff
@@ -332,6 +333,59 @@ class JavascriptParser extends Parser {
 					.setRange(expr.range);
 			}
 		});
+		this.hooks.evaluate.for("NewExpression").tap("JavascriptParser", _expr => {
+			const expr = /** @type {NewExpressionNode} */ (_expr);
+			const callee = expr.callee;
+			if (
+				callee.type !== "Identifier" ||
+				callee.name !== "RegExp" ||
+				expr.arguments.length > 2
+			)
+				return;
+
```

Comment:
You should check if somebody redefined `RegExp`.

`this.getVariableInfo("RegExp") === "RegExp"`

To test this, create a directory with `wrong` and `ok` and:

``` js
const RegExp = function() { return /other/; };

expect(require("./directory/" + ("ok".replace(new RegExp("ok"), "wrong")))).toBe("ok");
```

At runtime this will not replace ok with wrong, but if the evaluator incorrectly evaluates it to wrong it's no longer checked at runtime.